src/main/scala/PSO/Particle.scala

```diff
@@ -4,19 +4,21 @@ package pso
 import akka.actor._
 
 sealed trait ParticleMessage
-case object CalculateNewPosition extends ParticleMessage
+case object ComputeIteration extends ParticleMessage
 case class UpdateNeighbourhoodBest(best: List[Double]) extends ParticleMessage
 case class UpdateNeighbourhood(neighbours: Set[ActorRef]) extends ParticleMessage
+case class NewNeighbour(neighbour: ActorRef) extends ParticleMessage
 case object Report extends ParticleMessage
 
-class Particle(id: Int, initVelocity: List[Double], inertiaW: Double, accelCoeff1: Double, 
-    accelCoeff2: Double, randCoeff1: Double, randCoeff2: Double, lbest: List[Double], 
-    nbest: List[Double], objectiveFunction: List[Double] => Double) extends Actor {
-  val w = inertiaW
-  val c1 = accelCoeff1
-  val c2 = accelCoeff2
-  val r1 = randCoeff1
-  val r2 = randCoeff2
+class Particle(initVelocity: List[Double], inertia: Double, localAccel: Double, 
+    neighbourhoodAccel: Double, localR: Double, neighbourhoodR: Double, 
+    lbest: List[Double], nbest: List[Double], problemType: ProblemType.Value, 
+    objectiveFunction: List[Double] => Double) extends Actor {
+  val w = inertia
+  val c1 = localAccel
+  val c2 = neighbourhoodAccel
+  val r1 = localR
+  val r2 = neighbourhoodR
```

Comment:
https://github.com/alexandru/scala-best-practices/blob/master/sections/5-actors.md

5.2 regarding best practices of mutable state in actors